### ISSUE TITLE
incorrect date format code in ods and xls

### ISSUE THREAD

**Issue by aunriz** [PHPOffice/PhpSpreadsheet#592]
This is:

- [X] a bug report

### What is the expected behavior?

Reading dates from the cell having number format as YYYY-MM-DD from ods and xls documents, resulting in the date format code
`YYYY\-MM\-DD`

### What is the current behavior?
`d-mmm-yy h:mm:ss`

### What are the steps to reproduce?

The dates are in YYYY-MM-DD format using cell number formatting, when I save the sheets in ods and xls formats I get incorrect date format and when I save the sheets in xlsx format the date format is correct, below is the code that I am using to read spreadsheets

```php
<?php

require __DIR__ . '/vendor/autoload.php';

$PHPSpreadsheet = \PhpOffice\PhpSpreadsheet\IOFactory::load($file);
foreach ($PHPSpreadsheet->getWorksheetIterator() as $worksheet) {
    foreach ($worksheet->getRowIterator() as $row) {
        $cellIterator = $row->getCellIterator();
        $cellIterator->setIterateOnlyExistingCells(false); // Loop all cells, even if it is not set
        foreach ($cellIterator as $cell) {
            if (!is_null($cell)) {
                if (\PhpOffice\PhpSpreadsheet\Shared\Date::isDateTime($cell)) {
                    echo $cell->getStyle()->getNumberFormat()->getFormatCode();
                    echo '<br>';
                    echo $cell->getValue();
                    echo '<br>';
                    echo $cell->getCalculatedValue();
                    echo '<br>';
                    echo $cell->getFormattedValue();
                    echo '<br>';

                }
            }
        }
    }
}
```
this is the output that I get for xlsx, ods, xls respectively

```
YYYY\-MM\-DD
43196
43196
2018-04-06

d-mmm-yy h:mm:ss
43196.041666667
43196.041666667
6-Apr-18 1:00:00

d-mmm-yy h:mm:ss
43196.041666667
43196.041666667
6-Apr-18 1:00:00
```

### Which versions of PhpSpreadsheet and PHP are affected?
PHPSpreadsheet 1.2, PHP 7

**Comment by PowerKiKi** [comment]
Upgrade to latest version and report back whether it was fixed or not.

**Comment by aunriz** [comment]
Updated PHPSpreadsheet to 1.3.1, no change

**Comment by incubusSuccubus** [comment]
same problem, solution = set dates in excel with YYYY/MM/DD format

**Comment by Gopinath001** [comment]
bump,

**Comment by wergio** [comment]
in current version 1.10.1 is fixed only for XLS, but NOT for ODS!

**Comment by oleibman** [comment]
Resolved by PR #4806 in Feb. 2026. Leaving issue closed, removing stale label.
